docs/inventories.rst

```diff
@@ -0,0 +1,773 @@
+
+  [account_4321]
+  host3
+  host4 state=shutdown
+
+  [account_1234:vars]
+  account_alias=product_dev
+
+  [account_4321:vars]
+  account_alias=sustaining
+
+
+The goal here is to return only shutdown hosts that are present in the group with the ``account_alias`` variable of ``product_dev``.
+There are two approaches to this, both shown in yaml format. The first one suggested is recommended.
+
+.. _constr_inv_example:
+
+1. **Construct 2 groups, limit to intersection**
+
+``source_vars``:
+
+::
+
+  plugin: constructed
+  strict: true
+  groups:
+    is_shutdown: state | default("running") == "shutdown"
+    product_dev: account_alias == "product_dev"
+
+
+``limit``: ``is_shutdown:&product_dev``
+
+This constructed inventory input creates a group for both of the categories and uses the ``limit`` (host pattern) to only return hosts that are in the intersection of those two groups, which is documented in `host patterns in Ansible <https://docs.ansible.com/ansible/latest/inventory_guide/intro_patterns.html>`_.
+
+Also, when a variable may or may not be defined (depending on the host), you can give a default, like with ``| default("running")`` if you know what value it should have when it is not defined. This helps with debugging, as described in the :ref:`constr_inv_debugging` section.
+
+2. **Construct 1 group, limit to group**
+
+``source_vars``:
+
+::
+
+  plugin: constructed
+  strict: true
+  groups:
+    shutdown_in_product_dev: state | default("running") == "shutdown" and account_alias == "product_dev"
+
+
+``limit``: ``shutdown_in_product_dev``
+
+This input creates one group that only includes hosts that match both criteria. The limit is then just the group name by itself, returning just **host2**, same as the previous approach.
+
+.. _constr_inv_debugging:
+
+Debugging tips
+^^^^^^^^^^^^^^
+
+It is very important to set the ``strict`` parameter to ``True`` so that you can debug problems with your templates. If the template fails to render, you will get an error in the associated inventory update for that constructed inventory.
+
+When encountering errors, increase verbosity to get more details.
+
+Giving a default, like with ``| default("running")`` is a generic use of Jinja2 templates in Ansible. Doing this avoids errors from the particular template when you set ``strict: true``. You could also set ``strict: false``, and allow the template to produce an error, which results in the host not getting included in that group. However, doing so makes it difficult to debug issues in the future if your templates continue to grow in complexity.
+
+However, you may still have to debug the intended function of the templates if they are not producing the expected inventory content. For example, if a ``groups`` group has a complex filter (like ``shutdown_in_product_dev``) but does not contain any hosts in the resultant constructed inventory, then use the ``compose`` parameter to help debug. Like this:
+
+``source_vars``:
+
+::
```

Comment:
lexer needs to be set

Reply:
Do you mean apply a directive? I've been doing:
`.. envvar:: source_vars`

Is it acceptable?

Reply:
Looks like `YAML+Jinja` to me.

Reply:
Thanks for the suggestions. Since tests are passing, and this work can be separated out as an independent task, let's revisit this when I am more familiar with what these elements are you want applied and put all the code block changes in separate PR at a later time.